src/Microsoft.AspNetCore.HttpsPolicy/Microsoft.AspNetCore.HttpsPolicy.csproj

```diff
@@ -11,6 +11,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Rewrite\Microsoft.AspNetCore.Rewrite.csproj" />
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel.Core" Version="$(MicrosoftAspNetCoreServerKestrelCorePackageVersion)" />
```

Comment:
:s We don't want to do this. You need to extract the logic into a sources package

Reply:
Http.Sys has a similar though simpler parser. There may be a chance to unify all three.
https://github.com/aspnet/HttpSysServer/blob/14aec22cda2c0c81b369b08c7d260afffae20c9d/src/Microsoft.AspNetCore.Server.HttpSys/UrlPrefix.cs

Reply:
I would prefer this. Would it be possible to move all Url parsing to HttpAbstractions? cc/ @halter73

Reply:
Go for it.